src/samplers/adapt_sampler.js

```diff
@@ -0,0 +1,116 @@
+// @flow
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under
+// the License.
+
+import getInstanceId from './get_instance_id';
+import { SAMPLER_API_V2 } from './constants';
+import Span from '../span';
+import Utils from '../util';
+
+type AdaptSamplerFn = {
+  (sampler: any): ?Sampler,
+  orThrow: (sampler: any) => Sampler,
+};
+
+function adaptSamplerFn(sampler: any): ?Sampler {
+  if (!sampler) {
+    return null;
+  }
+  if (sampler.apiVersion === SAMPLER_API_V2) {
+    // already v2 API compatible
+    return sampler;
+  }
+  if (!sampler.apiVersion) {
+    // v1 legacy sampler
+    return new LegacySamplerV1Adapter(sampler);
+  }
+  return null;
+}
+
+function adaptSamplerOrThrow(sampler: any): Sampler {
+  const s = adaptSampler(sampler);
+  if (!s) {
+    throw new Error(`Unrecognized sampler: ${sampler}`);
+  }
+  return s;
+}
+
+adaptSamplerFn.orThrow = adaptSamplerOrThrow;
+const adaptSampler: AdaptSamplerFn = adaptSamplerFn;
+export default adaptSampler;
+
+class LegacySamplerV1Adapter implements Sampler {
+  apiVersion = SAMPLER_API_V2;
+
+  _adaptee: LegacySamplerV1;
```

Comment:
I thought we were going to keep Sampler interface unchanged for backwards compatibility and introduce SamplerV2.

Reply:
Per conversation with @yurishkuro, will leave as is.